homeassistant/components/tplink/number.py

```diff
@@ -0,0 +1,121 @@
+"""Support for TPLink number entities."""
+
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from kasa import Device, Feature
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import (
+    CoordinatedTPLinkFeatureEntity,
+    TPLinkFeatureEntityDescription,
+    async_refresh_after,
+    entities_for_device_and_its_children,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TPLinkNumberEntityDescription(
+    NumberEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+NUMBER_DESCRIPTIONS: Final = (
+    TPLinkNumberEntityDescription(
+        key="smooth_transition_on",
+        mode=NumberMode.BOX,
+    ),
+    TPLinkNumberEntityDescription(
+        key="smooth_transition_off",
+        mode=NumberMode.BOX,
+    ),
```

Comment:
The `NumberMode` is not working as I would expect (see the screenshot), any ideas?

Reply:
The value looks right. It might be a frontend bug

Reply:
matches how we do it in BAF

https://github.com/home-assistant/core/blob/08fae5d4197de891b998a5853fca98097d886a62/homeassistant/components/baf/number.py#L51